apps/hash-ai-worker-ts/src/activities/flow-activities/research-entities-action/infer-entities-from-web-page-worker-agent.ts

```diff
@@ -832,26 +673,110 @@ export const inferEntitiesFromWebPageWorkerAgent = async (params: {
 
-            const newProposedEntities = outputs.find(
-              ({ outputName }) =>
-                outputName ===
-                ("proposedEntities" satisfies OutputNameForAction<"inferEntitiesFromContent">),
-            )?.payload.value as ProposedEntity[];
+            const editionProvenance: ProposedEntity["provenance"] = {
+              actorType: "ai",
+              // @ts-expect-error - `ProvidedEntityEditionProvenanceOrigin` is not being generated correctly from the Graph API
+              origin: {
+                type: "flow",
+                id: flowEntityId,
+                stepIds: [activityId],
+              } satisfies OriginProvenance,
+            };
+
+            const propertyProvenance: NonNullable<
+              ProposedEntity["propertyMetadata"]
+            >[number]["metadata"]["provenance"] =
+              "url" in toolCallInput
+                ? {
+                    sources: [
+                      {
+                        type: SourceType.Webpage,
+                        location: {
+                          uri: toolCallInput.url,
+                          /** @todo */
+                          name: undefined,
+                          description: undefined,
+                        },
+                        loadedAt: new Date().toISOString(),
+                        /** @todo */
+                        authors: undefined,
+                        firstPublished: undefined,
+                        lastUpdated: undefined,
+                      },
+                    ],
+                  }
+                : {
+                    sources: [
+                      {
+                        type: SourceType.Document,
+                        location: {
+                          uri: toolCallInput.fileUrl,
+                          /** @todo */
+                          name: undefined,
+                          description: undefined,
+                        },
+                        loadedAt: accessedRemoteFile?.loadedAt,
+                        /** @todo */
+                        authors: undefined,
+                        firstPublished: validAt,
+                        lastUpdated: undefined,
+                      },
+                    ],
+                  };
+
+            const newProposedEntities = (
+              outputs.find(
+                ({ outputName }) =>
+                  outputName ===
+                  ("proposedEntities" satisfies OutputNameForAction<"inferEntitiesFromContent">),
+              )?.payload.value as ProposedEntity[]
+            ).map(
+              (proposedEntity) =>
+                ({
+                  ...proposedEntity,
+                  provenance: editionProvenance,
+                  propertyMetadata: Object.keys(proposedEntity.properties).map(
+                    (propertyBaseUrl) => ({
+                      /** @todo: figure out whether this is the path */
+                      path: [propertyBaseUrl],
```

Comment:
Obsolete comment?

Reply:
Yes, thanks for catching. Removed in https://github.com/hashintel/hash/pull/4399/commits/66d0d8c6a3c71159276273dd5d6049715a58a0d4